litestar/handlers/base.py

```diff
@@ -131,33 +134,66 @@ def __init__(
         self.middleware = middleware
         self.name = name
         self.opt = dict(opt or {})
+        self.opt.update(**kwargs)
         self.owner: Controller | Router | None = None
         self.return_dto = return_dto
-        self.signature_model: type[SignatureModel] | None = None
         self.signature_namespace = signature_namespace or {}
+        self.type_decoders = type_decoders
+        self.type_encoders = type_encoders
+
         self.paths = (
             {normalize_path(p) for p in path}
             if path and isinstance(path, list)
             else {normalize_path(path or "/")}  # type: ignore
         )
-        self.opt.update(**kwargs)
-        self.type_encoders = type_encoders
-        self.type_decoders = type_decoders
 
     def __call__(self, fn: AsyncAnyCallable) -> Self:
         """Replace a function with itself."""
         self._fn = Ref["MaybePartial[AsyncAnyCallable]"](fn)
         return self
 
+    @property
+    def handler_id(self) -> str:
+        """A unique idenrtifier used for generation of DTOs and schemas."""
+        return self.name or str(self)
```

Comment:
I think this might be an issue in cases where the same handler is registered multiple times but inherits different configs from upper layers. Maybe we should somehow take the ownership layers into account here and / or ensure that explicitly given names are globally unique?

Reply:
Fair point.

Reply:
done